fabric-api-base/src/main/java/net/fabricmc/fabric/api/event/EventFactory.java

```diff
@@ -47,6 +47,10 @@ public static void invalidate() {
 
 	/**
 	 * Create an "array-backed" Event instance.
+	 * The factory will be used to create the invoker for any number of listeners.
+	 *
+	 * <p>Consider using {@linkplain #createArrayBacked(Class, Object, Function) a custom empty invoker}
+	 * if performance of this event is critical.
```

Comment:
After your change, the empty invoker version seems to be less efficient.

Reply:
It is more efficient because the factory is invoked only in the event `update()` function. So in this case the resulting `emptyInvoker` is returned directly and stored in the event, and then returned directly when `.invoke()` is called.

Note that standard use passes `listeners -> (event signature) -> { ... }`, whereas here this is only `listeners -> /* some logic */`.

Reply:
Hmm, might need to specify. Like if the function passed in does not have an optimiztion for `listeners.length == 0` or `listeners.length == 1`, then this version is better. Otherwise, it isn't.

Reply:
Good idea, will update the javadoc in a bit.